### PR TITLE
feat: rabbitmq to check n messages with std props and json bodies

### PR DESCRIPTION
**Pull request by mariolg**
Two new steps are added:

1.  I wait up to "X" seconds for "Y" rabbit messages with the standard properties

2.  the body of the rabbit message in position "N" has the JSON properties

### CODE REVIEW

File: test/acceptance/features/rabbit.feature [hunk 1]
```diff
@@ -27,7 +27,7 @@ Feature: Rabbit client
           | id       | abc    |
           | name     | Golium |
 
-  @rabbit
+  @rabbit1
```
Comment (cbrea) [inline test/acceptance/features/rabbit.feature:30]: Restore tag

File: steps/rabbit/session.go [hunk 2]
```diff
@@ -311,9 +318,13 @@ func (s *Session) ValidateMessageTextBody(ctx context.Context, expectedMsg strin
 	return nil
 }
 
-// ValidateMessageJSONBody checks if the message json body properties are equal the expected values.
-func (s *Session) ValidateMessageJSONBody(ctx context.Context, props map[string]interface{}) error {
-	m := golium.NewMapFromJSONBytes([]byte(s.msg.Body))
+// ValidateMessageJSONBody checks if the message json body properties of message in position 'pos' are equal the expected values.
```
Comment (cbrea) [inline steps/rabbit/session.go:321]: In comment change "equal" by "equal to"

File: test/acceptance/features/rabbit.feature [hunk 3]
```diff
@@ -45,6 +45,36 @@ Feature: Rabbit client
           | id       | abc    |
           | name     | Golium |
 
+  @rabbit
+  Scenario: Publish and subscribe three JSON message. Use standard properties
```
Comment (cbrea) [inline test/acceptance/features/rabbit.feature:49]: Typo: message by messages

**Review by cbrea — APPROVED** [review]
Great work, just a minor comments

File: steps/rabbit/session.go [hunk 4]
```diff
@@ -254,18 +254,25 @@ func matchMessage(msg string, expectedProps map[string]interface{}) bool {
 	return true
 }
 
-// WaitForMessageWithStandardProperties waits for a message with standard rabbit properties that are equal to the expected values.
-func (s *Session) WaitForMessageWithStandardProperties(ctx context.Context, timeout time.Duration, props amqp.Delivery) error {
+// WaitForMessagesWithStandardProperties waits for 'count' messages with standard rabbit properties that are equal to the expected values.
+func (s *Session) WaitForMessagesWithStandardProperties(ctx context.Context, timeout time.Duration, count int, props amqp.Delivery) error {
 	return waitUpTo(timeout, func() error {
+		n := count
```
Comment (mbenitod) [inline steps/rabbit/session.go:260]: The variable "n" is not really needed, is it?
The rest of the things I was commenting on, cesar has included in his review.

**Review by mbenitod — CHANGES_REQUESTED** [review]
Have a look at my comment

File: steps/rabbit/session.go (hunk 4 above)
Comment (mariolg) [reply steps/rabbit/session.go:260]: Why not? is the variable been decreased, isn't it?

File: steps/rabbit/session.go (hunk 2 above)
Comment (mariolg) [reply steps/rabbit/session.go:320]: Done

File: steps/rabbit/session.go (hunk 4 above)
Comment (mariolg) [reply steps/rabbit/session.go]: Understood. Done

File: steps/rabbit/session.go (hunk 2 above)
Comment (mariolg) [reply steps/rabbit/session.go:320]: Done

File: test/acceptance/features/rabbit.feature (hunk 1 above)
Comment (mariolg) [reply test/acceptance/features/rabbit.feature]: Done

File: test/acceptance/features/rabbit.feature (hunk 3 above)
Comment (mariolg) [reply test/acceptance/features/rabbit.feature]: Done

**Pull request merged**